tests/test_mdlp.py

```diff
@@ -32,165 +34,31 @@ def test_find_cut_no_cut():
     k = find_cut(y, 0, len(y))
     assert_equal(-1, k)
 
-def test_mdlp_iris():
-    from sklearn.datasets import load_iris
+def test_fit_transform():
+  X = np.array([
```

Comment:
Can you rebase over my most recent commit? You should be able to pass in nested lists now.

Reply:
i did so and added a test case for passing in raw arrays. This even uncovered some unfortunate existing behavior where integer arrays were not being cast to np.float64 and this was relevant for the resulting discretization. I changed the check_array to also have a dtype for the data.